src/keywrap/mod.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright The ocicrypt Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::DecryptConfig;
+use crate::config::EncryptConfig;
+use anyhow::Result;
+use std::collections::HashMap;
+
+/// KeyWrapper is the interface used for wrapping keys using
+/// a specific encryption technology (pgp, jwe, pkcs7, pkcs11, keyprovider)
+pub trait KeyWrapper {
+    /// wrap keys data with encrypt config
+    fn wrap_keys(&self, ec: &EncryptConfig, opts_data: &[u8]) -> Result<Vec<u8>>;
+
+    /// unwrap keys data with decrypt config
+    fn unwrap_keys(&self, dc: &DecryptConfig, annotation: &[u8]) -> Result<Vec<u8>>;
+
+    /// get the keywraper annotation id
+    fn get_annotation_id(&self) -> &str;
+
+    /// no_possible_keys returns true if there is no possibility of performing
+    /// decryption for parameters provided.
+    fn no_possible_keys(&self, dc_param: &HashMap<String, Vec<Vec<u8>>>) -> bool;
+
+    /// get_private_keys (optional) gets the array of private keys. It is an optional implementation
+    /// as in some key services, a private key may not be exportable (i.e. HSM)
+    /// If not implemented, return the nil
+    fn get_private_keys(&self, dc_param: &HashMap<String, Vec<Vec<u8>>>) -> Option<Vec<Vec<u8>>>;
```

Comment:
There should be a default implementation for that one, that simply returns `None`.